src/proxy/http/add_header.rs

```diff
@@ -77,7 +77,7 @@ where
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         f.debug_struct("Layer")
             .field("header", &self.header)
-            .field("retrieve", &format_args!("{}", "..."))
+            .field("get_header", &format_args!("{}", "..."))
```

Comment:
i'll just note that this my complaint with adding debug impls to types we never try to format as debug... it's just more code that has to be maintained when we make small changes.

While we're here.. what's the deal with the format_args macro? We can't use "..." as a literal here?

Reply:
It's because `get_header` is not going to format to something helpful since it is a function pointer. It's comparable to `fmt::Debug` for [`Resolver`](https://github.com/linkerd/linkerd2-proxy/blob/6cc755881994cba967619a0999dd7b2339c986e2/src/dns.rs#L195). I'm not pointing this out as a reason to keep it; I think it proves your point more.

I'm okay removing the debug impl since it already does not serve a purpose in the code.

Reply:
@olix0r the `format_args!` was my suggestion --- we're not using a literal because it will be formatted using its `Debug` implementation, and the `Debug` implementation for strings includes the quotation marks, so it'll be formatted like this: `"..."`. However, formatting a string with `Display` omits the quotes. 

I thought we should avoid having the quotes so that the formatted output doesn't imply that the `get_header` field's value is a string.

With that said, I'm happy to agree that types which will not be formatted don't need `Debug` impls --- in the past we've generally tried to adhere to a practice of providing `Debug` impls for all types?

Reply:
@hawkw thanks, makes sense.